pandas/core/series.py

```diff
@@ -511,7 +511,10 @@ def _init_dict(
         elif index is not None:
             # fastpath for Series(data=None). Just use broadcasting a scalar
             # instead of reindexing.
-            values = na_value_for_dtype(pandas_dtype(dtype), compat=False)
+            if len(index) or dtype is not None:
```

Comment:
Do we know that `index` is list-like by the time we get here?

Reply:
Yes, partly from the type hints in this method and partly because of the `index = ensure_index(index)` part in the `__init__` method.